src/pick.c

```diff
@@ -446,6 +448,14 @@ selected_choice(void)
 			while (cursor_position < query_length
 			    && isu8cont(query[++cursor_position]));
 			break;
+		case PAGE_DOWN:
+			if (yscroll + lines - 1 < choices_count)
```

Comment:
That's a great improvement!

When I try it out locally it still feels a bit unnatural, though. For example, if I select the third choice by pressing the down arrow key twice and then press page up, nothing happens. What I'd expect is for the first choice to be selected. Another example is if I select the last choice in a list with more choices than would fit two pages, and then press page up it will advance the selection by two pages - 1.

I think what confuses me is that it is not immediately clear to me what the page up/down keys are controlling. In the case of the arrow keys, it is clearly controlling the selection and then the choices are scrolled to always show the selected one.

What do you think about letting the page up/down keys also control the selection by advancing it by a `lines` number of choices and then scrolling the choices to show the selection?

If `selection < lines` then it could `selection = 0` and when `choices_count - selection < lines` then it could `selection = choices_count`.